### ISSUE TITLE
"alarms" persistence flag affects netdata cloud

### ISSUE THREAD

**Issue by cakrit** [netdata/helmchart#93]
This Github issue is synchronized with Zendesk:

**Ticket ID:** [#20](https://netdata.zendesk.com/agent/tickets/20)
**Group:** Support
**Requester:** rahimian.pegah@gmail.com
**Assignee:** Zack
**Issue escalated by:** Christopher Akritidis

**Original ticket description:**

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div>Hi,</div><div>I have trouble signing in to my netdata cloud.</div><div>I am using the master-slave functionality of netdata. But I cannot see the nodes properly and slave nodes are getting disconnected.</div><div><br></div><div>Would you please kindly help me how can I solve this issue?</div><div><br></div><div>Regards,</div><div>Pegah<br></div></div></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-28 at 14:03:*

<blockquote><div class="zd-comment" dir="auto">Hi,<br>You probably have a setting in your browser that prevents the  required cookie from being stored. If you copy the magic link URL, and  paste it manually in a new tab in your browser with the developer  console showing, you will probably see an error. Please give us a  screenshot of any such errors and the browser version,  so we can help further. <br><br>Thanks,<br>Zack<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-04-28 at 14:19:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div>Thanks for your reply,</div><div>Now I can sign in with github. But also there are too many nodes there and the worker 1 , 2, 3, cannot be recognizes. That is relly strange as I cannot access the slaves in netdata. It happened after I turned on TLS. <br></div><div>Would you kindly help me how can I resolve this issue? <br></div><div><br></div><div>Thanks,</div><div>Pegah<br></div></div><br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-28 at 14:32:*

<blockquote><div class="zd-comment" dir="auto">Please report a bug in Github here <a href="https://github.com/netdata/netdata/issues/new/choose" rel="noreferrer">https://github.com/netdata/netdata/issues/new/choose</a> and provide as much detail as you can about your systems and setup.<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-28 at 14:32:*

<blockquote><div class="zd-comment" dir="auto"><span><a href="/users/403723624913" class="atmention">Christopher Akritidis</a></span>​ is this related to this one bug about masters/slaves and not seeing slave nodes?<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Christopher Akritidis on 2020-04-29 at 10:12:*

<blockquote><div class="zd-comment" dir="auto">Having it happen when TLS is turned on is key. I expect that the URL was http and now it should be https. So he could perhaps fix it by deleting some URLs from his visited nodes.<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Christopher Akritidis on 2020-04-29 at 10:13:*

<blockquote><div class="zd-comment" dir="auto">I can show you exactly what I mean, if you want.<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Costa Tsaousis on 2020-04-30 at 07:45:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr">Has anyone answered this?</div><br><div><div dir="ltr">On Tue, Apr 28, 2020 at 12:21 PM Pegah Rahimian &lt;<a href="mailto:rahimian.pegah@gmail.com" rel="noreferrer">rahimian.pegah@gmail.com</a>&gt; wrote:<br></div><blockquote style="border-left-color: rgb(204,204,204); border-left-style: solid; border-left-width: 1px; margin: 0px 0px 0px 0.8ex; padding-left: 1ex"><div dir="ltr"><div>Hi,</div><div>I have trouble signing in to my netdata cloud.</div><div>I am using the master-slave functionality of netdata. But I cannot see the nodes properly and slave nodes are getting disconnected.</div><div><br></div><div>Would you please kindly help me how can I solve this issue?</div><div><br></div><div>Regards,</div><div>Pegah<br></div></div>
</blockquote></div><br><div><br></div>-- <br><div dir="ltr"><div dir="ltr"><div><div dir="ltr"><div><div dir="ltr"><div dir="ltr"><div dir="ltr"><div dir="ltr"><div dir="ltr"><div dir="ltr"><div dir="ltr"><b><br></b></div><div dir="ltr"><b>Costa Tsaousis</b><br></div><div dir="ltr"><div>Founder &amp; CEO</div><div><br></div><div><img src="https://docs.google.com/uc?export=download&amp;id=1XvhQLOZDpOgS63ioL7JpCKs2jkLFT76i&amp;revid=0B5NLRecl4fQyY1Rpak0wNXoyaDBMYVpmeGpNaTVEenlvdmpRPQ" width="96" height="28"></div><div>Monitor everything&nbsp; ...in real-time!<br></div><div><b><a href="https://www.netdata.cloud" rel="noreferrer">https://www.netdata.cloud</a></b><br></div><div><br></div><div><font color="#999999">GR Mobile: </font>+306945494510 (<font color="#999999">also on @WhatsApp, @Viber, @Telegram)</font></div><div><font color="#999999">US Mobile: </font><font color="#000000">+1-(650)-885-8371</font><br><a href="https://www.linkedin.com/in/costatsaousis/" rel="noreferrer">LinkedIn profile</a>,&nbsp;<a href="https://github.com/ktsaou" rel="noreferrer">ktsaou @github</a>,&nbsp;ktsaou @skype</div><div><a href="https://www.linkedin.com/in/costatsaousis/" rel="noreferrer"><br></a></div></div></div></div></div></div></div></div></div></div></div></div></div></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-30 at 13:17:*

<blockquote><div class="zd-comment" dir="auto">Yes, and I have not heard back yet.</div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-30 at 17:30:*

<blockquote><div class="zd-comment" dir="auto">Hi Pegah,<br><br>When you turn on TLS you may have to check your visited nodes and make sure you are connecting to them using https:// instead of http:// since the old URLs will no longer be reachable.<br><br>Please let us know if you are still having this issue!<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-04-30 at 17:41:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div>Hi Zack,</div><div>Thanks, but I have deeper issue. I can see a lot of unreachable nodes which are replicating without any reason. As you see in the below screenshot:</div><div><div><img src="https://netdata.zendesk.com/attachments/token/2zgC1DrvnMPoTpvbFlTgyFQTs/?name=image.png" alt="image.png" width="495" height="251"><br></div><div>and then, wrong machine guide for worker 1 and worker 3:</div><div><div><img src="https://netdata.zendesk.com/attachments/token/hW4JNLnLyRg3rAMw7oX1rlu1K/?name=image.png" alt="image.png" width="476" height="465"><br></div><div>and unreachable for worker 2:</div><div><div>on the other hand, I only netdata-master-0 and worker 1, and worker 3 on the frontend. <br></div><div>I also receive this log from worker 2: <br></div><div>"<span>netdata ERROR : STREAM_SENDER[worker-2] : Cannot resolve host 'netdata', port '19999': Try again"</span></div><div><span><br></span></div><div><span>this behavior is not really normal. I also poster my issue, but they couldnt help me. <br></span></div><div><span>I would be so much appreciated if you help me or give me the hint for this problem. <br></span></div><div><span><br></span></div><div><span>Thanks a lot,</span></div><div><span>Pegah<br></span></div></div></div></div></div><br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-04-30 at 17:42:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div><img src="https://netdata.zendesk.com/attachments/token/7DijoBRyaj0ZosPPp29Mw0aZU/?name=image.png" alt="image.png" width="472" height="385"><br><br></div></div><br><div><div dir="ltr">On Thu, 30 Apr 2020 at 19:41, Pegah Rahimian &lt;<a href="mailto:rahimian.pegah@gmail.com" rel="noreferrer">rahimian.pegah@gmail.com</a>&gt; wrote:<br></div><blockquote style="border-left-color: rgb(204,204,204); border-left-style: solid; border-left-width: 1px; margin: 0px 0px 0px 0.8ex; padding-left: 1ex"><div dir="ltr"><div>Hi Zack,</div><div>Thanks, but I have deeper issue. I can see a lot of unreachable nodes which are replicating without any reason. As you see in the below screenshot:</div><div><div><img src="https://netdata.zendesk.com/attachments/token/2zEiCoE12aufI2pF7elEIglY2/?name=image.png" alt="image.png" width="495" height="251"><br></div><div>and then, wrong machine guide for worker 1 and worker 3:</div><div><div><img src="https://netdata.zendesk.com/attachments/token/vb1P4ArJO1IXuQLsucH5AgI1S/?name=image.png" alt="image.png" width="476" height="465"><br></div><div>and unreachable for worker 2:</div><div><div>on the other hand, I only netdata-master-0 and worker 1, and worker 3 on the frontend. <br></div><div>I also receive this log from worker 2: <br></div><div>"<span>netdata ERROR : STREAM_SENDER[worker-2] : Cannot resolve host 'netdata', port '19999': Try again"</span></div><div><span><br></span></div><div><span>this behavior is not really normal. I also poster my issue, but they couldnt help me. <br></span></div><div><span>I would be so much appreciated if you help me or give me the hint for this problem. <br></span></div><div><span><br></span></div><div><span>Thanks a lot,</span></div><div><span>Pegah<br></span></div></div></div></div></div><br>
</blockquote></div></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-04-30 at 17:43:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr">There is no TLS. I just have this issue without TLS setting.<br></div><br><div><div dir="ltr">On Thu, 30 Apr 2020 at 19:42, Pegah Rahimian &lt;<a href="mailto:rahimian.pegah@gmail.com" rel="noreferrer">rahimian.pegah@gmail.com</a>&gt; wrote:<br></div><blockquote style="border-left-color: rgb(204,204,204); border-left-style: solid; border-left-width: 1px; margin: 0px 0px 0px 0.8ex; padding-left: 1ex"><div dir="ltr"><div><img src="https://netdata.zendesk.com/attachments/token/Wd623MPOqdwOMdRMMr2h2hici/?name=image.png" alt="image.png" width="472" height="385"><br><br></div></div><br><div><div dir="ltr">On Thu, 30 Apr 2020 at 19:41, Pegah Rahimian &lt;<a href="mailto:rahimian.pegah@gmail.com" rel="noreferrer">rahimian.pegah@gmail.com</a>&gt; wrote:<br></div><blockquote style="border-left-color: rgb(204,204,204); border-left-style: solid; border-left-width: 1px; margin: 0px 0px 0px 0.8ex; padding-left: 1ex"><div dir="ltr"><div>Hi Zack,</div><div>Thanks, but I have deeper issue. I can see a lot of unreachable nodes which are replicating without any reason. As you see in the below screenshot:</div><div><div><img src="https://netdata.zendesk.com/attachments/token/ujO2UPJ7wm4xipKZieB0nUYh3/?name=image.png" alt="image.png" width="495" height="251"><br></div><div>and then, wrong machine guide for worker 1 and worker 3:</div><div><div><img src="https://netdata.zendesk.com/attachments/token/VuFiyHkaYia0OdbYxFRW8vpEE/?name=image.png" alt="image.png" width="476" height="465"><br></div><div>and unreachable for worker 2:</div><div><div>on the other hand, I only netdata-master-0 and worker 1, and worker 3 on the frontend. <br></div><div>I also receive this log from worker 2: <br></div><div>"<span>netdata ERROR : STREAM_SENDER[worker-2] : Cannot resolve host 'netdata', port '19999': Try again"</span></div><div><span><br></span></div><div><span>this behavior is not really normal. I also poster my issue, but they couldnt help me. <br></span></div><div><span>I would be so much appreciated if you help me or give me the hint for this problem. <br></span></div><div><span><br></span></div><div><span>Thanks a lot,</span></div><div><span>Pegah<br></span></div></div></div></div></div><br>
</blockquote></div>
</blockquote></div></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-30 at 17:47:*

<blockquote><div class="zd-comment" dir="auto"><a href="https://github.com/netdata/netdata/issues/8847" rel="noreferrer">https://github.com/netdata/netdata/issues/8847</a><br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-30 at 17:53:*

<blockquote><div class="zd-comment" dir="auto">ignore, i think the prev linked issue is not the right one (and i can't edit that comment)<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-30 at 17:55:*

<blockquote><div class="zd-comment" dir="auto">If you made a Github issue, can you link me that issue?<br><br>Also, what is your Netdata version? You can get that with "netdata -v"<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-04-30 at 17:57:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div>netdata v1.20.0-278-nightly</div><div><br></div><div>and here is the issue:</div><div><br></div><div><a href="https://github.com/netdata/netdata/issues/8847#event-3280654319" rel="noreferrer">https://github.com/netdata/netdata/issues/8847#event-3280654319</a></div><div><br></div></div><br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-04-30 at 18:01:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div>probably you can access UI?</div><div><br></div><div><a href="http://152.66.245.138:22647/host/master-from-snapshot/#menu_system_submenu_idlejitter;theme=slate;help=true" rel="noreferrer">http://152.66.245.138:22647/host/master-from-snapshot/#menu_system_submenu_idlejitter;theme=slate;help=true</a></div><div><br></div><div>although we are reinstalling k8s in few hours to check if it will solve the issue<br></div></div><br><div><div dir="ltr">On Thu, 30 Apr 2020 at 19:57, Pegah Rahimian &lt;<a href="mailto:rahimian.pegah@gmail.com" rel="noreferrer">rahimian.pegah@gmail.com</a>&gt; wrote:<br></div><blockquote style="border-left-color: rgb(204,204,204); border-left-style: solid; border-left-width: 1px; margin: 0px 0px 0px 0.8ex; padding-left: 1ex"><div dir="ltr"><div>netdata v1.20.0-278-nightly</div><div><br></div><div>and here is the issue:</div><div><br></div><div><a href="https://github.com/netdata/netdata/issues/8847#event-3280654319" rel="noreferrer">https://github.com/netdata/netdata/issues/8847#event-3280654319</a></div><div><br></div></div><br>
</blockquote></div></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-30 at 19:19:*

<blockquote><div class="zd-comment" dir="auto">I cannot access the UI.<br><br>My current version is netdata v1.21.1 - do you have an update available?<br><br>I went through the github issue.&nbsp; When you reinstall k8s are you going to redeploy/update your helm charts?<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-04-30 at 20:03:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div>I also tried v21. But same issue.</div><div>for reinstalling, I will use "helm delete netdata", then: " helm install netdata ./path_to_hem_chart"</div><div>&nbsp;</div><div>I assume it should be delete and purge as Im using helm3. But you as you see in the screenshot, I can see a lot of previous unreachable nodes. I dont understand how those nodes are replicated.</div><div><br></div><div>Thanks,</div><div>Pegah<br></div></div><br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-04-30 at 20:17:*

<blockquote><div class="zd-comment" dir="auto">Ok. Let me know when you reinstall, then we can add some details to the github issue if that doesn't fix it.<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-04-30 at 20:19:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div><div dir="auto">Thanks a lot for your help. I will contact you tomorrow,</div></div><div dir="auto"><br></div><div dir="auto">Pegah</div><div><br></div></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Christopher Akritidis on 2020-05-01 at 11:19:*

<blockquote><div class="zd-comment" dir="auto">I know where the issue is coming from, will write on GitHub.<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Christopher Akritidis on 2020-05-01 at 12:05:*

<blockquote><div class="zd-comment" dir="auto">Ok, I saw the discussion in 8847 and it has to do with streaming, which is unrelated to this report. I will unlink it, so it doesn't get confusing.<br><br>The issue here is caused by an confusing link between the "alarm persistence" and the netdata instance unique identifier (MACHINE_GUID), which is not apparent in the documentation for values.yaml. `/var/lib/netdata` is used to store both the alarms log and the `registry` information (specifically netdata.registry.unique.id).&nbsp; So we'll need to update our helm chart, to make it much clearer. We'll possibly need to force /var/lib/netdata to always use persistent storage, but that will be discussed in an issue.&nbsp;<br><br>What you can do from your side as a workaround, to prevent new netdata masters from appearing is to turn the master alarms persistence flag to true and provide a storageclass that will work in your environment.&nbsp;<br></div></blockquote>

**Comment by cakrit** [comment]
We basically have three options here:
- Provide more options for persistence, that distinguish between the various files stored under `/var/lib/netdata`. A `claim.d` directory is getting added there as well. I'm not fond of this option, because it means we'll need to always add new options and require new volumes every time we add a new file or directory there.
- Change the docs and the config files so that it's clear it's not just `alarms` that supposedly use the persistent storage for `/var/lib/netdata` but also key things like the machine guid, the claiming information and the health management API key. We strongly recommend that users persist this volume and only disable it if they don't want to use netdata cloud.
- Just demand at least one persistent volume, to be used for `/var/lib/netdata`, removing completely the option to disable it.

Would like to hear your views.

**Comment by cakrit** [comment]
cc @pegy4444

**Comment by cakrit** [comment]
cc @peggy4444

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-05-03 at 19:26:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div>Hi,</div><div>Thanks for checking. We have re-installed k8s, and installed netdata with helm chart (v21) again. I have also set the persistent flags for alarm and database to true. But we still cannot access the slaves immediately. There are only 2 nodes: "netdata-master-0 &amp; master"&nbsp; So we still have streaming problem for slaves. And this is the log I receive from all workers:</div><div><br></div><div><span>netdata ERROR : STREAM_SENDER[worker-2] : Cannot resolve host 'netdata', port '19999': Try again (errno 22, Invalid argument)</span></div><div><span><br></span></div><div><span>Would you please kindly help me how to solve the streaming issue? Does it have to anything with localStorages?<br></span></div><div><span><a href="http://152.66.245.138:22647" rel="noreferrer">http://152.66.245.138:22647</a></span></div></div><br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-05-03 at 20:57:*

<blockquote><div class="zd-comment" dir="auto">I cannot access the address you provided, is it internal?<br><br>The helm chart was updated recently (<a href="https://github.com/netdata/helmchart/blob/master/Chart.yaml#L4" rel="noreferrer">1.2.0</a> for 1.21.0).<br><br>For now I made a new helm chart issue here: <a href="https://github.com/netdata/helmchart/issues/94" rel="noreferrer">https://github.com/netdata/helmchart/issues/94</a> in case it's a problem with how the services are set up.<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-05-04 at 15:34:*

<blockquote><div class="zd-comment" dir="auto">Hi Pegah,<br><br>Just to keep you updated: We've decided to continue the discussion in issue <a href="https://github.com/netdata/netdata/issues/8847" rel="noreferrer">8847</a>. There are a few things for us to test out - thank you for reporting this, you are helping make Netdata better!<br>Have a look at the latest comments and let me know if some of those suggestions work for you.<br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-05-04 at 21:05:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div><div dir="auto">Hi,</div></div><div dir="auto">Thanks, here is what we see:</div><div dir="auto"><br></div><div dir="auto">1) This issue is not concerning us as we are not using netdata cloud. So thats not a big deal. I have just asked you as it was un normal to me.</div><div dir="auto"><br></div><div dir="auto">2) &nbsp;we want to encrypt intra-cluster communication, but we could not set TLS master-slave successfully yet</div><div dir="auto"><br></div><div dir="auto">That will be great if you let me know the result of your testing, as we are still in trouble of streaming configuration, and I don’t have any idea how to reach out all the slaves after enabling TLS. Are you able to do that while your testing phase?</div><div dir="auto"><br></div><div dir="auto">Thanks,</div><div dir="auto">Pegah</div><div><br></div></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-05-04 at 22:21:*

<blockquote><div class="zd-comment" dir="auto">I will let you know once our testing is complete. If we run into unexpected problems, the discussion will be in the issue I linked previously.<br></div></blockquote>

**Comment by Peggy4444** [comment]
Ok thanks,
That will be great if you just describe the result of testing, regardless
of being successful or not, and sharing configuration in the issue page.

Thanks again and regards,
Pegah

**Comment by cakrit** [comment]
*Comment made from Zendesk by Pegah Rahimian on 2020-05-19 at 22:54:*

<blockquote><div class="zd-comment zd-comment-pre-styled" dir="auto"><div dir="ltr"><div>Dear Zack,</div><div><br></div><div>Is there someone who I can ask how to enable weblog module in netdata helm chart? <br></div><div>I want to see upstream_response_time chart in netdata. (probably for nginx).</div><div>I have enabled go.d plugin in values.yaml, but still no upstream_response_time chart in UI.</div><div>Do I need any other configuration?<br></div><div><br></div><div>Thanks again,</div><div>Pegah<br></div></div><br></div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-05-20 at 07:29:*

<blockquote><div class="zd-comment" dir="auto">You need to configure both nginx and Netdata and make sure to restart both. Also see https://learn.netdata.cloud/docs/agent/collectors/python.d.plugin/nginx/<br><br>I will look into the specific value  and the helm chart and let you know.</div></blockquote>

**Comment by cakrit** [comment]
*Comment made from Zendesk by Zack on 2020-05-20 at 16:50:*

<blockquote><div class="zd-comment" dir="auto"><a href="https://github.com/netdata/helmchart/issues/100" rel="noreferrer">https://github.com/netdata/helmchart/issues/100</a><br><br></div></blockquote>

**Comment by cakrit** [comment]
Closing in favor of #114
